src/LinAlg/hiopMatrixSparseTriplet.cpp

```diff
@@ -925,13 +925,13 @@ void hiopMatrixSparseTriplet::convertToCSR(int &csr_nnz, int **csr_kRowPtr_in, i
   double *csr_kVal = *csr_kVal_in;
   int *csr_jCol = *csr_jCol_in;
 
-  int *index_covert_extra_Diag2CSR_temp = new int[n];
+  int *index_covert_extra_Diag2CSR_temp = new int[n]{};
   int *nnz_each_row_tmp = new int[n]{};
 
   // set correct col index and value
   {
-    *index_covert_CSR2Triplet_in = new int[csr_nnz];
-    *index_covert_extra_Diag2CSR_in = new int[n];
+    *index_covert_CSR2Triplet_in = new int[csr_nnz]{};
```

Comment:
I do not think the extra initialization to 0 (`{}`) is needed as you are rewriting all the elements in a loop later on...

Reply:
they are not necessary. Just try to keep consistence with the other part, e.g., line 929 in your reference above.

Reply:
unnecessary extra work is being done, please remove